src/main/errors/Error.ts

```diff
@@ -0,0 +1,18 @@
+class BaseError extends Error {
+  public name: string;
+  public status: number;
+
+  constructor(errorName: string, errorMessage: string, status: number) {
+    super(errorMessage);
+    this.name = errorName;
+    this.status = status;
+  }
+}
```

Comment:
Ooh I like this abstraction, nice job!

Reply:
What's the reason behind separating the two files, `libraries/Error.ts` and `errors/Error.ts`?

Reply:
I wanted `libraries/Error.ts` to be more readable and we can customize the message (or other details in the future)- so it's kind of like a helper/factory file that uses errors from the `errors/Error.ts` class!

Reply:
Hmm I see, I think devs may be confused whether to use the Error.ts from libraries or the one from errors. For me I'm not too sure which one to use when looking at it.

Imo I like the idea of the `errors/Error.ts` class, I find that's sufficient enough. What we could do is have an `errors/ErrorMessages.ts` and do something like:

```
class ForbiddenError extends BaseError {
  constructor(errorMessage?: string) {
    super('ForbiddenError', errorMessage ?? ErrorMessages.INSUFFICIENT_PRIVELEGES, 403);
  }
}

// then we can just call
throw new ForbiddenError();
```

The benefit of this is that all errors are contained in one file and more simplicity since there's less confusion between libraries/errors directory.

Reply:
Hmm yeah I can see how you can get confused! I'll update to something similar to what you suggested here!